### ISSUE TITLE
receiving port is in use

### ISSUE THREAD

**Issue by abunezek** [colinbdclark/osc.js#207]
how to solve this problem of EADDRINUSE when trying to receive data when the port is use by the program who's sending it? 
I've tried using an OSC monitor and it can open this port with no problem even when its in use,
why osc.js can't receive from this port when its in use?

**Comment by colinbdclark** [comment]
Hi @abunezek, sorry to hear you're having difficulty. It's hard to provide any advice without knowing more about what you're trying to do. Which transport are you using? Can you create a simplified test case that illustrates the problem you're seeing?

**Comment by abunezek** [comment]
Hi, and thanks for your response.
Let me elaborate on the issue:
I've developed a program called "j-counter," and you can find it [here ](https://www.jcounter.net/jcounter)(a demo is available for testing). There's a control program named 'bitfocus companion' that utilizes OSC.js to receive and send OSC data to and from other programs, including mine. I've created a module for the companion program to communicate with my application, and it needs to display timer data based on OSC commands received from it.

The problem arises when I start my program, (written in C# and using rug.osc). It opens a port for sending OSC data. While other OSC-receiving programs, like[ this one](https://hexler.net/protokol), work without any issues, OSC.js sends an error message stating that the port is already in use (EADDRINUSE).
As far as I know  The essence of OSC, which relies on UDP, is that one program can send data over a specific port, and others over the network can receive it simultaneously. Unfortunately, this is not happening in this case.

here is my code:
```
this.osc = new OSC.UDPPort({
localPort: this.config.feedbackPort,
localAddress: '0.0.0.0',
remoteAddress: this.config.host,
remotePort: this.config.port,
broadcast: true,
metadata: false,
})

this.osc.on('ready', () => {
})
this.osc.on('message', (message) => this.onMessage(message))
this.osc.on('error', (err) => {
this.log('info', err.code)
if (err.code == 'EADDRINUSE') {
this.log('error', `Error: Selected feedback port ${err.message.split(':')[1]} is already in use.`)
this.updateStatus('bad_config', 'Feedback port conflict')
}
})
this.osc.open()

this.log('info', `osc_server_init method finished ${this.osc}`)
}
```

In summary, I need OSC.js to open a port and receive OSC even when the port is in use on the sending program and any other listening programs.

Here are some links:
1. my program: https://www.jcounter.net/jcounter (a demo can be downloaded for testing)
2. The modules full code: https://github.com/bitfocus/companion-module-jozeemedia-jcounter/blob/main/main.js
3. 
Thank you so much for your assistance.

**Comment by colinbdclark** [comment]
Hi @abunezek. First, I'm not sure what you mean by "the essence of OSC" relying on UDP. Open Sound Control is a transport-agnostic protocol, and UDP is one of several transports supported by osc.js. It uses Node.js' UDP implementation directly for its UDP transport, so you can have a look at Node.js' documentation if you need a primer on how UDP works in this environment.

The situation you describe should be quite possible with osc.js, and I know that many other users have successfully implemented an architecture like this. I'm not really in a position to provide networking support for commercial software products at this time. If you think there's a bug in osc.js, please provide a simplified test case that I can run directly on my Mac without additional dependencies, and I'm happy to take a look.

Good luck with your project!
